mmedit/models/diffusers/ddpm_diffuser.py

```
@@ -0,0 +1,195 @@
+        self.num_inference_steps = num_inference_steps
+        self.timesteps = np.arange(
+            0, self.num_train_timesteps,
+            self.num_train_timesteps // self.num_inference_steps)[::-1].copy()
+
+    def _get_variance(self, t, predicted_variance=None, variance_type=None):
+        alpha_prod_t = self.alphas_cumprod[t]
+        alpha_prod_t_prev = self.alphas_cumprod[t - 1] if t > 0 else self.one
+
+        # For t > 0, compute predicted variance βt (see formula (6) and (7) from https://arxiv.org/pdf/2006.11239.pdf) # noqa
+        # and sample from it to get previous sample
+        # x_{t-1} ~ N(pred_prev_sample, variance) == add variance to pred_sample # noqa
+        variance = (1 - alpha_prod_t_prev) / (1 - alpha_prod_t) * self.betas[t]
+
+        if t == 0:
+            log_variance = (1 - alpha_prod_t_prev) / (
+                1 - alpha_prod_t) * self.betas[1]
+        else:
+            log_variance = np.log(variance)
+
+        if variance_type is None:
+            variance_type = self.variance_type
+
+        # hacks - were probs added for training stability
+        if variance_type == 'fixed_small':
+            variance = np.clip(variance, min_value=1e-20)
+        # for rl-diffuser https://arxiv.org/abs/2205.09991
+        elif variance_type == 'fixed_small_log':
+            variance = np.log(np.clip(variance, min_value=1e-20))
+        elif variance_type == 'fixed_large':
+            variance = self.betas[t]
+        elif variance_type == 'fixed_large_log':
+            # Glide max_log
+            variance = np.log(self.betas[t])
+        elif variance_type == 'learned':
+            return predicted_variance
+        elif variance_type == 'learned_range':
+            min_log = log_variance
+            max_log = np.log(self.betas[t])
+            frac = (predicted_variance + 1) / 2
+            log_variance = frac * max_log + (1 - frac) * min_log
+            variance = torch.exp(log_variance)
+
```

Please raise an error when `variance_type` is not supported.